guides/error_handling.md

```diff
@@ -0,0 +1,39 @@
+# Error Handling
+
+This page guides you through handling errors in Oban.
+
+The basics: jobs can fail in expected or unexpected ways. To mark a job as failed, you can return `{:error, reason}` from a worker's [`perform/1` callback](`c:Oban.Worker.perform/1`), as documented in the `t:Oban.Worker.result/0` type. A job can also fail because of unexpected raised errors or exits.
+
+In any case, when a job fails the details of the failure are recorded in the `errors` array on the `Oban.Job` struct.
+
+## Error Details
+
+Oban stores execution errors as a list of maps (`t:Oban.Job.errors/0`). Each error contains the following keys:
+
+  * `:at` — The UTC timestamp when the error occurred at
+  * `:attempt` — The attempt number when the error occurred
+  * `:error` — A *formatted* error message and stacktrace
+
+See the [Instrumentation docs](instrumentation.html) for an example of
+integrating with external error reporting systems.
+
+## Retries
+
+When a job fails and the number of execution attempts is below the configured
+`max_attempts` limit for that job, the job will automatically be retried in the future.
```

Comment:
This would be a good place to mention that the job is `discarded` when failed attempts matches `max_attempts`.